core/OpenGL/UniformProvider.h

```diff
@@ -61,7 +62,7 @@ class UniformProvider{
 		}
 	}
 
-	void setUniform(std::string name, const Vector3& value) {
+	void setUniform(std::string name, const Vector4& value) {
```

Comment:
Unjustified change of function. Will also result in compiler errors, because of conflicting signatures.